src/zip_code_tree.hpp

```diff
@@ -68,140 +74,96 @@ class ZipCodeTree {
 
 
       A snarl in the vector is bounded by a SNARL_START and a SNARL_END.
       A snarl is comprised of the two bounds, one or more chains, and the distances among them.
       SEEDs are always contained within a chain.
-      For each element of the snarl (boundary or child chain), the distance to each element preceding
-      it in the snarl is stored before the element.
+      For each element of the snarl (boundary or child chain), the distance to each element 
+      preceding it in the snarl is stored before the element.
       The distances are stored in reverse order of the elements that they reach.
-      Immediately before the SNARL_END, there is a NODE_COUNT storing the number of children in the snarl
-      A snarl would look like:
+      Immediately before the SNARL_END, there is a NODE_COUNT storing the number of children in the
+      snarl. A snarl would look like:
           SNARL_START, dist:start->c1, chain1, dist:c1->c2, dist:start->c2, chain2, ..., 
                 ..., dist:c2->end, dist:c1->end, dist:start->end, node_count, SNARL_END
-      For snarls that aren't dags (called cyclic snarls, even though they could have an inversion and 
-      no cycles), all seeds on the snarl are split up into mini chains comprised of seeds that are
-      on the same chain with no seeds on snarls between them. In order to represent all edges between
-      all pairs of node sides, each chain is represented multiple times. Each chain is represented first
-      in its forward orientation (which is arbitrary), immediately followed by a copy in the reverse 
-      orientation. All chains are then repeated in both orientations a second time
-
-
-      Everything is ordered according to the order of the highest-level chain (top-level chain or child
-      of a top-level snarl).
-      For children of a snarl, the children are ordered according to a topological sort of the snarl.
-      In the variation graph, all chains are considered to be oriented "forward" in their parent snarl.
-      However, in a start-to-end traversal of the snarl, the child chain may be traversed end-to-start.
-      These chains would be considered to be reversed in the zip code tree, so the order of the children
-      of the chain may be backwards relative to their order in the variation graph.
-      If a snarl is the child of a chain that is traversed backwards in the zip tree, then that snarl
-      and all its children are also traversed backwards.
+
+      For snarls that aren't dags (called cyclic snarls, even though they could have an inversion 
+      and no cycles), the zip tree should represent all possible paths that the read could take 
+      through the snarl. All seeds on the snarl are split up into "runs" of seeds on the same chain
+      that are "close" to each other. The runs are sorted and orientated by their read coordinate 
+      and each run is made into a separate child chain like normal. A run occur twice, once in 
```

Comment:
```suggestion
      and each run is made into a separate child chain like normal. A run occurs twice, once in 
```

Are we doubling the run even if the non-dag snarl is a cyclic directed graph and not actually bidirected?